src/reflect/scala/reflect/internal/Names.scala

```diff
@@ -470,6 +470,29 @@ trait Names extends api.Names {
     def append(separator: Char, suffix: Name) = newName(toString + separator + suffix)
     def prepend(prefix: String) = newName("" + prefix + this)
 
+    def stripSuffix(suffix: String): ThisNameType =
+      if (endsWith(suffix)) dropRight(suffix.length) else thisName // OPT avoid creating a Name with `suffix`
+
+    def stripSuffix(suffix: Name): ThisNameType =
+      if (endsWith(suffix)) dropRight(suffix.length) else thisName
+
+    def take(n: Int): ThisNameType      = subName(0, n)
+    def drop(n: Int): ThisNameType      = subName(n, length)
+    def dropRight(n: Int): ThisNameType = subName(0, length - n)
+
+    def dropLocal: TermName      = toTermName stripSuffix NameTransformer.LOCAL_SUFFIX_STRING
+    def dropSetter: TermName     = toTermName stripSuffix NameTransformer.SETTER_SUFFIX_STRING
+    def dropModule: ThisNameType = this stripSuffix NameTransformer.MODULE_SUFFIX_STRING
+    def localName: TermName      = getterName append NameTransformer.LOCAL_SUFFIX_STRING
+    def setterName: TermName     = getterName append NameTransformer.SETTER_SUFFIX_STRING
+    def getterName: TermName     = dropTraitSetterSeparator.dropSetter.dropLocal
+
+    private def dropTraitSetterSeparator: TermName =
+      indexOf(NameTransformer.TRAIT_SETTER_SEPARATOR_STRING) match {
+        case -1  => toTermName
+        case idx => toTermName drop idx drop NameTransformer.TRAIT_SETTER_SEPARATOR_STRING.length
```

Comment:
It may also be more efficient to join the two `drop`: 
```suggestion
        case idx => toTermName drop (idx + NameTransformer.TRAIT_SETTER_SEPARATOR_STRING.length)
```